org.eclipse.buildship.core.test/src/main/groovy/org/eclipse/buildship/core/workspace/internal/SynchronizingBuildScriptUpdateListenerTest.groovy

```diff
@@ -0,0 +1,70 @@
+package org.eclipse.buildship.core.workspace.internal
+
+import org.eclipse.buildship.core.configuration.BuildConfiguration
+import org.eclipse.buildship.core.configuration.WorkspaceConfiguration
+import org.eclipse.buildship.core.test.fixtures.ProjectSynchronizationSpecification
+import org.eclipse.buildship.core.test.fixtures.WorkspaceSpecification
+
+import java.io.File
+
+import org.eclipse.core.resources.IProject
+import org.eclipse.core.runtime.NullProgressMonitor
+import org.eclipse.jdt.core.IClasspathEntry
+import org.eclipse.jdt.core.IJavaProject
+
+class SynchronizingBuildScriptUpdateListenerTest extends ProjectSynchronizationSpecification {
```

Comment:
I've implemented one test so that you can finish the class. 
```
    WorkspaceConfiguration workspaceConfig

    def setup() {
        workspaceConfig = configurationManager.loadWorkspaceConfiguration()
    }

    def cleanup() {
        configurationManager.saveWorkspaceConfiguration(workspaceConfig)
    }
    def "Execute project synchronization when build.gradle file created"() {
        setup:
        File projectDir = dir('auto-sync-test-project') {
            dir('src/main/java')
        }
        importAndWait(projectDir)
        IProject project = findProject('auto-sync-test-project')
        enableAutoSyncForProject(project)

        when:
        String buildScript = '''
            apply plugin: "java"
            repositories { jcenter() }
            dependencies { compile "org.springframework:spring-beans:1.2.8" }
        '''
        project.getFile('build.gradle').create(new ByteArrayInputStream(buildScript.bytes), false, new NullProgressMonitor())
        waitForResourceChangeEvents()
        waitForGradleJobsToFinish()

        then:
        JavaCore.create(project).getResolvedClasspath(false).find{ it.path.toPortableString().endsWith('spring-beans-1.2.8.jar') }
    }

    private void enableAutoSyncForWorkspace() {
        configurationManager.saveWorkspaceConfiguration(new WorkspaceConfiguration(workspaceConfig.gradleDistribution, workspaceConfig.gradleUserHome,
                                                                                   workspaceConfig.gradleIsOffline, workspaceConfig.buildScansEnabled,
                                                                                   true))
    }

    private void enableAutoSyncForProject(IProject project) {
        BuildConfiguration currentConfig = configurationManager.loadProjectConfiguration(project).buildConfiguration;
        BuildConfiguration updatedConfig = configurationManager.createBuildConfiguration(currentConfig.getRootProjectDirectory(),
            true,
            currentConfig.gradleDistribution,
            currentConfig.gradleUserHome,
            currentConfig.buildScansEnabled,
            currentConfig.offlineMode,
            true)
        configurationManager.saveBuildConfiguration(updatedConfig)
    }
```